frontend/app/src/components/history/events/forms/EthWithdrawalEventForm.vue

```diff
@@ -0,0 +1,270 @@
+  {
+    editableItem: null,
+    groupHeader: null
+  }
+);
+
+const { t } = useI18n();
+
+const { editableItem, groupHeader } = toRefs(props);
+const { currencySymbol } = storeToRefs(useGeneralSettingsStore());
+
+const assetPriceForm: Ref<InstanceType<
+  typeof HistoryEventAssetPriceForm
+> | null> = ref(null);
+
+const datetime: Ref<string> = ref('');
+const amount: Ref<string> = ref('');
+const usdValue: Ref<string> = ref('');
+const validatorIndex: Ref<string> = ref('');
+const withdrawalAddress: Ref<string> = ref('');
+const isExit: Ref<boolean> = ref(false);
+
+const errorMessages = ref<Record<string, string[]>>({});
+
+const rules = {
+  timestamp: { externalServerValidation: () => true },
+  amount: {
+    required: helpers.withMessage(
+      t('transactions.events.form.amount.validation.non_empty').toString(),
+      required
+    )
+  },
+  usdValue: {
+    required: helpers.withMessage(
+      t('transactions.events.form.fiat_value.validation.non_empty', {
+        currency: get(currencySymbol)
+      }).toString(),
+      required
+    )
+  },
+  validatorIndex: {
+    required: helpers.withMessage(
+      t(
+        'transactions.events.form.validator_index.validation.non_empty'
+      ).toString(),
+      required
+    )
+  },
+  withdrawalAddress: {
+    required: helpers.withMessage(
+      t(
+        'transactions.events.form.withdrawal_address.validation.non_empty'
+      ).toString(),
+      required
+    ),
+    isValid: helpers.withMessage(
+      t(
+        'transactions.events.form.withdrawal_address.validation.valid'
+      ).toString(),
+      (value: string) => isValidEthAddress(value)
+    )
+  }
+};
+
+const { setValidation, setSubmitFunc } = useHistoryEventsForm();
+
+const v$ = setValidation(
+  rules,
+  {
+    timestamp: datetime,
+    amount,
+    usdValue,
+    validatorIndex,
+    withdrawalAddress
+  },
+  {
+    $autoDirty: true,
+    $externalResults: errorMessages
+  }
+);
+
+const reset = () => {
+  set(datetime, convertFromTimestamp(dayjs().unix()));
+  set(amount, '0');
+  set(usdValue, '0');
+  set(validatorIndex, '');
+  set(withdrawalAddress, '');
+  set(isExit, false);
+  set(errorMessages, {});
+
+  get(assetPriceForm)?.reset();
+};
+
+const applyEditableData = async (entry: EthWithdrawalEvent) => {
+  set(datetime, convertFromTimestamp(entry.timestamp));
+  set(amount, entry.balance.amount.toFixed());
+  set(usdValue, entry.balance.usdValue.toFixed());
+  set(validatorIndex, entry.validatorIndex.toString());
+  set(withdrawalAddress, entry.locationLabel);
+  set(isExit, entry.isExit);
+};
+
+const applyGroupHeaderData = async (entry: EthWithdrawalEvent) => {
+  set(withdrawalAddress, entry.locationLabel ?? '');
+  set(validatorIndex, entry.validatorIndex.toString());
+  set(datetime, convertFromTimestamp(entry.timestamp));
+};
+
+const { setMessage } = useMessageStore();
+
+const { editHistoryEvent, addHistoryEvent } = useHistoryTransactions();
+
+const save = async (): Promise<boolean> => {
```

Comment:
So question, how much of the save method, and check props repeats in different components? 

in save the only thing that seems different is the payload.

Do you think we can move the common part to a composable and only pass the pass that differs?